jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/util/WebSocketFlusher.java

```diff
@@ -0,0 +1,213 @@
+            else
+            {
+                _failure.addSuppressed(t);
+            }
+        }
+
+        if (failed)
+            _flusher.abort(t);
+    }
+
+    private class Flusher extends IteratingCallback implements Callback
+    {
+        private boolean _completed = false;
+        private OutgoingEntry _current;
+
+        @Override
+        protected Action process() throws Throwable
+        {
+            boolean first = false;
+            try (AutoLock l = _lock.lock())
+            {
+                if (_failure != null)
+                    throw _failure;
+
+                if (_current == null)
+                {
+                    first = true;
+                    _current = _entries.poll();
+                }
+            }
+
+            if (_current == null)
+                return Action.IDLE;
+
+            if (log.isDebugEnabled())
+                log.debug("onFrame {}", _current);
+
+            _completed = onFrame(new OutgoingEntry.Builder(_current).callback(this).build(), first);
+            return Action.SCHEDULED;
+        }
+
+        @Override
+        protected void onSuccess()
+        {
+            if (_completed)
+            {
+                notifyCallbackSuccess(_current);
+                _current = null;
+            }
+        }
+
+        @Override
+        protected void onCompleteFailure(Throwable t)
+        {
+            if (log.isDebugEnabled())
+                log.debug("onCompleteFailure {}", t.toString());
+
+            List<OutgoingEntry> entries;
+            try (AutoLock l = _lock.lock())
+            {
+                if (_failure == null)
+                    _failure = t;
+                entries = new ArrayList<>(_entries);
+                _entries.clear();
+            }
+
+            if (_current != null)
+            {
+                notifyCallbackFailure(_current, t);
+                _current = null;
+            }
+
+            for (OutgoingEntry entry : entries)
+                notifyCallbackFailure(entry, t);
+
+            // Allow any subclass to clean up internal state on failure.
+            WebSocketFlusher.this.onCompleteFailure(t);
+        }
+    }
+
+    private void notifyCallbackSuccess(OutgoingEntry entry)
+    {
+        if (log.isDebugEnabled())
+            log.debug("notifyCallbackSuccess {}", entry);
+
+        try
+        {
+            entry.getCallback().succeeded();
+        }
+        catch (Throwable x)
+        {
+            log.warn("Exception while notifying success of entry {}", entry, x);
+        }
+    }
+
+    private void notifyCallbackFailure(OutgoingEntry entry, Throwable failure)
+    {
+        if (log.isDebugEnabled())
+            log.debug("notifyCallbackFailure {} {}", entry, failure.toString());
+
+        try
+        {
+            entry.getCallback().failed(failure);
+        }
+        catch (Throwable x)
+        {
+            log.warn("Exception while notifying failure of entry {}", entry, x);
```

Comment:
LOG.info()

Reply:
Debug or warn.  info is for more expected stuff.

Reply:
Debug seems a bit light for this error.